### PR TITLE
[feature] Add bisectors method for Polygon class

### PR DESCRIPTION
**Pull request by namannimmo10**
#### References to other Issues or PRs

#12599

#### Release Notes

*   geometry
     *  Added `bisectors` method for `Polygon` class.

### CODE REVIEW

**Comment by namannimmo10** [comment]
@smichr Please look into this.

**Comment by namannimmo10** [comment]
I think the problem is definitely in comparison. Should I check `__eq__` methods of point and ray? 
Acc. to Travis' log:
```
Failed example:
    for k, v in p.bisectors(2).items():
        print(k, v)
Expected:
    (Point2D(0, 3), Ray2D(Point2D(0, 3), Point2D(0.23, 2.0)))
    (Point2D(0, 0), Ray2D(Point2D(0, 0), Point2D(0.71, 0.71)))
    (Point2D(2, 0), Ray2D(Point2D(2, 0), Point2D(1.1, 0.38)))
    (Point2D(1, 1), Ray2D(Point2D(1, 1), Point2D(0.19, 0.42)))
Got:
    Point2D(0, 3) Ray2D(Point2D(0, 3), Point2D(0.23, 2.0))
    Point2D(0, 0) Ray2D(Point2D(0, 0), Point2D(0.71, 0.71))
    Point2D(2, 0) Ray2D(Point2D(2, 0), Point2D(1.1, 0.38))
    Point2D(1, 1) Ray2D(Point2D(1, 1), Point2D(0.19, 0.42))

```

**Comment by oscarbenjamin** [comment]
That looks like the different Py2 and Py3 print function

**Comment by namannimmo10** [comment]
dict comparison seems okay. what has `print` got to do with this?

**Comment by sylee957** [comment]
If you just use python 2 print like `print(a, b)`, it would be similar as `print (a, b)` and internally considers it as 'Print a tuple' rather than 'Print a and b'

**Comment by namannimmo10** [comment]
I know `print (a, b)` will result in a tuple. But how do I fix this issue in py3?

**Comment by namannimmo10** [comment]
And why does it give an error? 
```
>>> p.bisectors(2)
{Point2D(0, 3): Ray2D(Point2D(0, 3), Point2D(0.23, 2.0)), Point2D(0, 0): Ray2D(Point2D(0, 0), Point2D(0.71, 0.71)), Point2D(2, 0): Ray2D(Point2D(2, 0), Point2D(1.1, 0.38)), Point2D(1, 1): Ray2D(Point2D(1, 1), Point2D(0.19, 0.42))}
>>> assert p.bisectors(2) == {
...     Point2D(0, 3): Ray2D(Point2D(0, 3), Point2D(0.23, 2.0)),
...     Point2D(0, 0): Ray2D(Point2D(0, 0), Point2D(0.71, 0.71)),
...     Point2D(2, 0): Ray2D(Point2D(2, 0), Point2D(1.1, 0.38)),
...     Point2D(1, 1): Ray2D(Point2D(1, 1), Point2D(0.19, 0.42))
... }
Traceback (most recent call last):
  File "<stdin>", line 5, in <module>
AssertionError
```

**Comment by oscarbenjamin** [comment]
> I know `print (a, b)` will result in a tuple. But how do I fix this issue in py3?

Don't use `print` in doctests.

> And why does it give an error?

Probably because of the floats. That isn't a nice way to write the docstring though. This should be fine:
```
>>> p.bisectors(2)
{Point2D(0, 3): Ray2D(Point2D(0, 3), Point2D(0.23, 2.0)),
 Point2D(0, 0): Ray2D(Point2D(0, 0), Point2D(0.71, 0.71)),
 Point2D(2, 0): Ray2D(Point2D(2, 0), Point2D(1.1, 0.38)),
 Point2D(1, 1): Ray2D(Point2D(1, 1), Point2D(0.19, 0.42))}
```

**Comment by namannimmo10** [comment]
Thanks for the suggestions, @oscarbenjamin.

**Comment by sylee957** [comment]
Try to replace like
```diff
$ git diff
diff --git a/sympy/geometry/polygon.py b/sympy/geometry/polygon.py
index e69a4279b3..522dfc51de 100644
--- a/sympy/geometry/polygon.py
+++ b/sympy/geometry/polygon.py
@@ -1382,10 +1382,10 @@ def bisectors(p, prec=None):
         >>> from sympy import Polygon, Point
         >>> p = Polygon(Point(0, 0), Point(2, 0), Point(1, 1), Point(0, 3))
         >>> p.bisectors(2)
-        {Point2D(0, 3): Ray2D(Point2D(0, 3), Point2D(0.23, 2.0)),
-         Point2D(0, 0): Ray2D(Point2D(0, 0), Point2D(0.71, 0.71)),
-         Point2D(2, 0): Ray2D(Point2D(2, 0), Point2D(1.1, 0.38)),
-         Point2D(1, 1): Ray2D(Point2D(1, 1), Point2D(0.19, 0.42))}
+        {Point2D(0, 0): Ray2D(Point2D(0, 0), Point2D(0.71, 0.71)),
+         Point2D(0, 3): Ray2D(Point2D(0, 3), Point2D(0.23, 2.0)),
+         Point2D(1, 1): Ray2D(Point2D(1, 1), Point2D(0.19, 0.42)),
+         Point2D(2, 0): Ray2D(Point2D(2, 0), Point2D(1.1, 0.38))}
         """
         b = {}
         pts = list(p.args)
```

**Comment by sylee957** [comment]
I think that some tests should be added to get the coverage rate up.

**Comment by namannimmo10** [comment]
I'm getting assertion error while comparing dicts. 
```
assert p.bisectors(2) == \
       {Point2D(0, 0): Ray2D(Point2D(0, 0), Point2D(0.71, 0.71)),
        Point2D(0, 3): Ray2D(Point2D(0, 3), Point2D(0.23, 2.0)),
        Point2D(1, 1): Ray2D(Point2D(1, 1), Point2D(0.19, 0.42)),
        Point2D(2, 0): Ray2D(Point2D(2, 0), Point2D(1.1, 0.38))}
```
> Traceback (most recent call last):
  File "/home/namannimmo/sympy/sympy/geometry/tests/test_polygon.py", line 415, in test_bisectors
    Point2D(2, 0): Ray2D(Point2D(2, 0), Point2D(1.1, 0.38))}
AssertionError

**Comment by namannimmo10** [comment]
Maybe it's because of float values in the dict. bc in general, we can compare dicts using `==`.

**Comment by namannimmo10** [comment]
@smichr, please suggest?

File: sympy/geometry/polygon.py [hunk 1]
```diff
@@ -1370,6 +1370,40 @@ def __contains__(self, o):
...
+        b = {}
+        pts = list(p.args)
+        pts.append(pts[0])  # close it
+        cw = Polygon._isright(*pts[:3])
+        if cw:
+            pts = list(reversed(pts))
+        for v, a in p.angles.items():
+            i = pts.index(v)
+            ray = Ray(pts[i], pts[i + 1]).rotate(a/2, v)
+            # normalize so distance from p1 to p2 is 1
```
[the remark this answers is not in the document]
Comment (namannimmo10) [reply sympy/geometry/polygon.py]: How do I normalize? Should I use `_normalize_dimension`? cause this code is written by @smichr.

**Comment by namannimmo10** [comment]
@sylee957 If any changes are still required, please suggest.. otherwise, I think this is ready now.

File: sympy/geometry/polygon.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (namannimmo10) [reply sympy/geometry/polygon.py]: Thanks!

**Comment by namannimmo10** [comment]
Please merge this.

**Pull request merged**